Use Train upper constraints on Python 2

priteau opened the pull request.

markgoddard commented on defaults/main.yml:
```diff
@@ -74,4 +74,8 @@ os_images_upload: True
+
+# Use Train upper constraints when running with Python 2, to avoid
+# incompatibility with newer OSC releases.
+os_images_upper_constraints_file: "{% if ansible_python.version.major == '2' %}https://releases.openstack.org/constraints/upper/train{% endif %}"
```
It's already on L9

priteau replied:
I didn't notice, thanks!

markgoddard commented on meta/main.yml:
```diff
@@ -26,9 +26,9 @@ dependencies:
     os_openstacksdk_venv: "{{ os_images_venv }}"
     os_openstacksdk_state: "{{ os_images_package_state }}"
-    os_openstacksdk_upper_constraints: "{{ os_images_upper_constraints }}"
+    os_openstacksdk_upper_constraints_file: "{{ os_images_upper_constraints }}"
```
Should it be os_images_upper_constraints_file?

priteau replied:
Good catch.

markgoddard commented on meta/main.yml:
```diff
@@ -26,9 +26,9 @@ dependencies:
     os_openstackclient_venv: "{{ os_images_venv }}"
     os_openstackclient_state: "{{ os_images_package_state }}"
-    os_openstackclient_upper_constraints: "{{ os_images_upper_constraints }}"
+    os_openstackclient_upper_constraints_file: "{{ os_images_upper_constraints }}"
```
ditto

priteau committed:
Use Train upper constraints on Python 2

Also fix variable names for openstacksdk and openstackclient upper
constraints: they were missing the _file suffix.
